QuizWhois.Domain/Services/Implementations/QuestionService.cs

```diff
@@ -30,42 +31,91 @@ public async Task<QuestionModel> AddQuestion(QuestionModel questionModel)
 
             var correctAnswers = new List<CorrectAnswer>();
             questionModel.CorrectAnswers.ForEach(x => correctAnswers.Add(new CorrectAnswer(x)));
-            var entity = new Question { QuestionText = questionModel.QuestionText, CorrectAnswers = correctAnswers };
+            var entity = new Question { QuestionText = questionModel.QuestionText, CorrectAnswers = correctAnswers, PackId = questionModel.PackId };
             var result = await _context.Questions.AddAsync(entity);
             await _context.SaveChangesAsync();
             _logger.LogInformation($"Question id = {entity.Id} was added");
             var correctAnswersToModel = new List<string>();
             result.Entity.CorrectAnswers.ForEach(x => correctAnswersToModel.Add(x.AnswerText));
-            return new QuestionModel(result.Entity.Id, result.Entity.QuestionText, correctAnswersToModel);
+            return new QuestionModelResponse(result.Entity.Id, result.Entity.QuestionText, correctAnswersToModel, result.Entity.PackId);
         }
 
-        public QuestionModel GetQuestion(long questionId)
+        public QuestionModelResponse GetQuestion(long questionId)
         {
-            if (questionId <= 0)
-            {
-                throw new Exception("Id was invalid number");
-            }
+            DataValidation.ValidateId(questionId);
 
             var entity = _context.Questions.Where(x => x.Id == questionId).Select(x => new
                 {
                     id = x.Id,
                     questionText = x.QuestionText,
-                    correctAnswers = x.CorrectAnswers
+                    correctAnswers = x.CorrectAnswers,
+                    packId = x.PackId
                 }).FirstOrDefault();
 
             var correctAnswers = new List<string>();
             entity.correctAnswers.ToList().ForEach(x => correctAnswers.Add(x.AnswerText));
-            return new QuestionModel(entity.id, entity.questionText, correctAnswers);
+            return new QuestionModelResponse(entity.id, entity.questionText, correctAnswers, entity.packId);
+        }
+
+        public async Task UpdateQuestion(QuestionModelRequest questionModel, long questionId)
+        {
+            DataValidation.ValidateId(questionId);
+
+            var entity = _context.Questions.Where(x => x.Id == questionId).Select(x => new Question
+            {
+                Id = x.Id,
+                QuestionText = x.QuestionText,
+                CorrectAnswers = x.CorrectAnswers,
+                PackId = x.PackId
+            }).FirstOrDefault();
```

Comment:
it is a bit excessive. you can use FirstOrDefault() here earlier, instead of Where()

Reply:
But then i can't use Select after FirstOrDefault